src/clients/chunker.rs

```diff
@@ -108,28 +125,32 @@ impl Client for ChunkerClient {
         "chunker"
     }
 
-    async fn health(&self) -> HealthCheckResult {
-        let mut client = self.health_client.clone();
-        let response = client
-            .check(HealthCheckRequest { service: "".into() })
-            .await;
+    async fn health(&self) -> Result<HealthCheckResult, Error> {
+        let request =
+            grpc_request_with_headers(HealthCheckRequest { service: "".into() }, HeaderMap::new());
+        let response: Result<HealthCheckResponse, Error> =
+            async { grpc_call!(self.health_client, request, HealthClient::check) }.await;
```

Comment:
TODO: undecided if `let request = async { grpc_call!(self.health_client, request, HealthClient::check) }.await;` is too messy or if i should modify the macro slightly to avoid needing the async block when not immediately returning the value of the `grpc_call!` in a handler function